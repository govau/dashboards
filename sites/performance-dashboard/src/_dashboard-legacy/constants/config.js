export const SCREEN_SM = 768;
export const DEFAULT_HERO_HEIGHT = 240;
