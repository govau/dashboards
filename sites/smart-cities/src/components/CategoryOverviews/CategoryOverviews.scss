@import '~style/main';

.footerSpacerWrapper {
  padding-top: 32px;
  padding-bottom: 32px;
  background: $color-white;

  @include for-tablet-portrait-up {
    padding-top: 68px;
    padding-bottom: 62px;
  }
}

.footerSpacerContainer {
  @include container;
}

.footerSpacerLine {
  background: $color-grey--200;
  height: 1px;
}

.footerSpacerShade {
  height: 32px;
  background: $color-primary--020;

  @include for-tablet-portrait-up {
    height: 64px;
  }
}
