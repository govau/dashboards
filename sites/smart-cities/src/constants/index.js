export * from './misc';
export { default as INDICATORS } from './indicators';
export * from './categories';
export { default as CITIES } from './cities';
