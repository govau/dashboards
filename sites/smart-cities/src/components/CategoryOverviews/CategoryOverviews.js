import React from 'react';
import PropTypes from 'prop-types';
import CategoryOverview from '../CategoryOverview/CategoryOverview';
import { CATEGORIES } from '../../constants';
import style from './CategoryOverviews.scss';

const CategoryOverviews = props => (
  <div>
    {CATEGORIES.map(category => (
      <CategoryOverview
        key={category.id}
        category={category}
        cities={props.cities}
        city={props.city}
      />
    ))}

    <div className={style.footerSpacerWrapper}>
      <div className={style.footerSpacerContainer}>
        <div className={style.footerSpacerLine} />
      </div>
    </div>

    <div className={style.footerSpacerShade} />
  </div>
);

const cityPropShape = {
  id: PropTypes.string.isRequired,
  indicators: PropTypes.object.isRequired,
};

CategoryOverviews.propTypes = {
  city: PropTypes.shape(cityPropShape),
  cities: PropTypes.arrayOf(PropTypes.shape(cityPropShape)),
};

export default CategoryOverviews;
