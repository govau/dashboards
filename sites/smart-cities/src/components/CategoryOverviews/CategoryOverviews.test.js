import React from 'react';
import { shallow } from 'enzyme';
import CategoryOverviews from './CategoryOverviews';
import { CATEGORIES } from '../../constants';

jest.mock('../CategoryOverview/CategoryOverview', () => 'CategoryOverview');

const defaultProps = {
  city: {
    id: 'perth',
    indicators: {}
  },
};

it('should match Snapshot', () => {
  const component = shallow(
    <CategoryOverviews {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should render the correct number of categories', () => {
  const component = shallow(
    <CategoryOverviews {...defaultProps} />
  );

  expect(component.find('CategoryOverview').length).toBe(CATEGORIES.length)
});
