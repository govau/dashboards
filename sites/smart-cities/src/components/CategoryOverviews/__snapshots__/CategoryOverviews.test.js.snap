// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div>
  <CategoryOverview category={{...}} cities={[undefined]} city={{...}} />
  <CategoryOverview category={{...}} cities={[undefined]} city={{...}} />
  <CategoryOverview category={{...}} cities={[undefined]} city={{...}} />
  <CategoryOverview category={{...}} cities={[undefined]} city={{...}} />
  <CategoryOverview category={{...}} cities={[undefined]} city={{...}} />
  <CategoryOverview category={{...}} cities={[undefined]} city={{...}} />
  <CategoryOverview category={{...}} cities={[undefined]} city={{...}} />
  <div className=\\"footerSpacerWrapper\\">
    <div className=\\"footerSpacerContainer\\">
      <div className=\\"footerSpacerLine\\" />
    </div>
  </div>
  <div className=\\"footerSpacerShade\\" />
</div>"
`;
